File: 백준/Bronze/1157. 단어 공부/단어 공부.c
#include <stdio.h>
#include <string.h>

int main()
{
    char s[1000000], answer;
    int arr[26], len, max = 0;
    scanf("%s", s);
    len = strlen(s);
    
    for (int i = 0; i < len; i++) {
        if (s[i] >= 'a') arr[s[i] - 'a']++;
        else arr[s[i] - 'A']++;
    }
    
    for (int i = 0; i < 26; i++) {
        if (arr[i] == max) answer = '?';
        else if (arr[i] > max) {
            max = arr[i];
            answer = 'A' + i;
        }
    }
    
    printf("%c", answer);
}

File: 백준/Bronze/1157. 단어 공부/단어 공부.java
import java.util.*;

public class Main
{
	public static void main(String[] args) {
	    Scanner sc = new Scanner(System.in);
	    
	    String s = sc.next();
	    int max = 0;
	    int[] arr = new int[26];
	    char answer = '?';
	    
	    for (int i = 0; i < s.length(); i++) {
	        if (s.charAt(i) >= 'a') arr[s.charAt(i) - 'a']++;
	        else arr[s.charAt(i) - 'A']++;
	    }
	    
	    for (int i = 0; i < 26; i++) {
	        if (arr[i] == max) answer = '?';
	        else if (arr[i] > max) {
	            max = arr[i];
	            answer = (char) ('A' + i);
	        }
	    }
	    
		System.out.print(answer);
	}
}